### PR TITLE
[luigi.six.getargspec] removed

### PR DESCRIPTION
**Pull request by drowoseque**
## Description
`luigi.six.getargspec` removed

## Motivation and Context
#2830

### CODE REVIEW

**Comment by drowoseque** [comment]
@honnix 
@dlstadther 
PTAL
thanks

File: luigi/scheduler.py [hunk 1]
```diff
@@ -96,12 +95,15 @@ def rpc_method(**request_args):
...
-
-        assert not fn_args.varargs
-        assert fn_args.args[0] == 'self'
-        all_args = fn_args.args[1:]
-        defaults = dict(zip(reversed(all_args), reversed(fn_args.defaults or ())))
+        args, varargs, varkw, defaults, kwonlyargs, kwonlydefaults, ann = inspect.getfullargspec(fn)
+        if kwonlyargs or ann:
+            raise ValueError("Function has keyword-only parameters or annotations"
+                             ", use getfullargspec() API which can support them")
+        (first_arg, *all_args) = args
```
Comment (dlstadther) [inline luigi/scheduler.py]: are these parentheses necessary? (genuine question, i'm unsure)

File: luigi/scheduler.py [hunk 2]
```diff
@@ -96,12 +95,15 @@ def rpc_method(**request_args):
...
+
+        assert not varargs
```
Comment (dlstadther) [inline luigi/scheduler.py]: Should this be checked before unpacking `args`?

Comment (drowoseque) [reply luigi/scheduler.py]: yes, makes sense
thanks

File: luigi/scheduler.py (hunk 1 above)
Comment (drowoseque) [reply luigi/scheduler.py]: no, they are not necessary
just it looks more clean with them

File: luigi/scheduler.py [hunk 3]
```diff
@@ -96,12 +95,15 @@ def rpc_method(**request_args):
...
         # the decorator function with the request args attached.
-        fn_args = six.getargspec(fn)
-
-        assert not fn_args.varargs
-        assert fn_args.args[0] == 'self'
-        all_args = fn_args.args[1:]
-        defaults = dict(zip(reversed(all_args), reversed(fn_args.defaults or ())))
+        args, varargs, varkw, defaults, kwonlyargs, kwonlydefaults, ann = inspect.getfullargspec(fn)
+        if kwonlyargs or ann:
+            raise ValueError("Function has keyword-only parameters or annotations"
```
Comment (honnix) [inline luigi/scheduler.py]: Not due to this PR, but I'm not sure what this error tries to tell. Here we already use `getfullargspec`. Do I miss anything?

Comment (drowoseque) [reply luigi/scheduler.py]: @honnix 
I think that's related to the fact, this is going on inside of function `getargspec` which is not supposed to deal with annotations and keyword-only parameters
so this exception seems to be some sort of code-consistency among python version
but I'm still unsure.
I think we can remove this raising

**Comment by drowoseque** [comment]
@honnix 
@dlstadther 
PTAL
thanks

**Comment by cianru** [comment]
@honnix
@dlstadther
ping

**Pull request merged**